### PR TITLE
feat(server): add new hook render:beforeResponse 

### LINKED ISSUE #7468: Can't overwrite Accept-Ranges header
### Version

[v2.12.2](https://github.com/nuxt.js/releases/tag/v2.12.2)

### Reproduction link

[https://github.com/nuxt/nuxt.js/blob/dev/packages/server/src/middleware/nuxt.js#L81](https://github.com/nuxt/nuxt.js/blob/dev/packages/server/src/middleware/nuxt.js#L81)

### Steps to reproduce

See code line

### What is expected ?

I happen to need to overwrite the Accept-Ranges header

### What is actually happening?

Can't do that because `res.setHeader('Accept-Ranges', 'none')` is called right before ` res.end(...)` is being called.

### Additional comments?

The line `res.setHeader('Accept-Ranges', 'none')` should be above the call for `render:route` hook.
In that why the header could be rewritten.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10735">#c10735</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by Dima000**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Resolves: #7468
It also allows to overwrite headers set by next renderer.

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/server/src/middleware/nuxt.js [hunk 1]
```diff
@@ -78,7 +79,6 @@ export default ({ options, nuxt, renderRoute, resources }) => async function nux
 
     // Send response
     res.setHeader('Content-Type', 'text/html; charset=utf-8')
-    res.setHeader('Accept-Ranges', 'none') // #3870
     res.setHeader('Content-Length', Buffer.byteLength(html))
     res.end(html, 'utf8')
     await nuxt.callHook('render:routeDone', url, result, context)
```
Comment (pi0) [inline packages/server/src/middleware/nuxt.js:84]: There might be other use-cases to override headers and while `render:route` hook  _can_ do it it is more for context modification not doing response logic. Maybe we can add a new `render:beforeResponse` hook before `res.end` or simply swap L83 with L84?

Comment (pi0) [reply packages/server/src/middleware/nuxt.js:84]: Second idea: we can use `res.getHeader` to ensure not overriding any header already set

Comment (Dima000) [reply packages/server/src/middleware/nuxt.js:84]: There was a similar discussion a year ago https://github.com/nuxt/nuxt.js/pull/4755
Maybe it is a good idea to go with the first option you mentioned @pi0

I will try to implement the required changes and see how it goes. Thanks for the reply!

**Review by pi0 — APPROVED** [review]

**Review by clarkdo — APPROVED** [review]

**Comment by pi0** [comment]
Thanks @Dima000 :heart:

**Comment by manniL** [comment]
Please don't forget to alter the documentation as well.

**Comment by Dima000** [comment]
I'll do a PR to the docs this week. 
Thanks for the feedback guys and thanks for mentaining this awesome project

**Pull request merged**